src/Component/Symfony/Maker/MakeResource.php

```diff
@@ -0,0 +1,206 @@
+use Sylius\Component\Resource\Metadata\Resource;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Resource\Symfony\Maker\Generator\RepositoryClassGenerator;
+use Sylius\Component\Resource\Symfony\Maker\Generator\ResourceClassGenerator;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Bundle\MakerBundle\Util\ClassDetails;
+use Symfony\Bundle\MakerBundle\Util\ClassSourceManipulator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+final class MakeResource extends AbstractMaker
+{
+    public function __construct(
+        private Generator $generator,
+        private ResourceClassGenerator $resourceClassGenerator,
+        private RepositoryClassGenerator $repositoryClassGenerator,
+        private FileManager $fileManager,
+    ) {
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function getCommandName(): string
+    {
+        return 'make:resource';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function getCommandDescription(): string
+    {
+        return 'Creates a Resource class';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->setDescription(self::getCommandDescription())
+            ->addArgument('name', InputArgument::OPTIONAL, 'Class name of the resource to create')
+            ->addOption('is-entity', null, InputOption::VALUE_NONE, 'Do you want to store resource data in the database (via Doctrine)?')
+            ->setHelp(file_get_contents(__DIR__ . '/Resources/help/MakeResource.txt'))
+        ;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function configureDependencies(DependencyBuilder $dependencies): void
+    {
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        $resourceIsEntity = $io->confirm(
+            'Do you want to store resource data in the database (via Doctrine)?',
+            class_exists(DoctrineBundle::class),
+        );
+
+        if (null === $input->getOption('is-entity')) {
+            $input->setOption('is-entity', $resourceIsEntity);
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator): void
+    {
+        /** @var string $name */
+        $name = $input->getArgument('name');
+
+        /** @var bool $resourceIsEntity */
```

Comment:
```suggestion
        /** @var bool|null $resourceIsEntity */
```

Maybe i's better to set a default value for this option to avoid getting null at this point, WDYT ?